Lesson_04/homework_04.py

```diff
@@ -0,0 +1,130 @@
+console_handler = logging.StreamHandler()
+console_handler.setFormatter(log_formatter)
+_log.addHandler(console_handler)
+_log.setLevel(logging.DEBUG)
+
+adventures_of_tom_sayer = """\
+Tom gave up the brush with reluctance in his .... face but alacrity
+in his heart. And while
+the late steamer
+"Big Missouri" worked ....
+and sweated
+in the sun,
+the retired artist sat on a barrel in the .... shade close by,
+dangled his legs,
+munched his apple, and planned the slaughter of more innocents.
+There was no lack of material;
+boys happened along every little while;
+they came to jeer, but .... remained to whitewash. ....
+By the time Ben was fagged out,
+Tom had traded the next chance to Billy Fisher for
+a kite, in good repair;
+and when he played
+out, Johnny Miller bought
+in for a dead rat and a string to swing it with—and so on, and so on,
+hour after hour. And when the middle of the afternoon came, from being a
+poor poverty, stricken boy in the .... morning, Tom was literally
+rolling in wealth."""
+
+#  ПЕРЕЗАПИСУЙТЕ зміст змінної adventures_of_tom_sayer у завданнях 1-3
+# task 01 ==
+# Дані у строці adventures_of_tom_sayer
+# розбиті випадковим чином, через помилку.
+# треба замінити кінець абзацу на пробіл .replace("\n", " ")
+# Replacing "\n" with spaces " ".
+new_adventures_of_tom_sayer = adventures_of_tom_sayer.replace('\n', ' ')
+_log.info(new_adventures_of_tom_sayer)
+
+# task 02 ==
+# Замініть .... на пробіл
+# Replacing "...." with spaces " ".
+new_adventures_of_tom_sayer = new_adventures_of_tom_sayer.replace('....', ' ')
+_log.info(new_adventures_of_tom_sayer)
+
+# Option 2 for task 01 and task 02
+# Create a translation table where "\n" = " " and  "...." = null
+mapping_table = adventures_of_tom_sayer.maketrans('\n', ' ', '....')
+# Apply the specified translations to the input string.
+tom_sayer = adventures_of_tom_sayer.translate(mapping_table)
+_log.info(tom_sayer)
+
+# task 03 ==
+# Зробіть так, щоб у тексті було не більше одного пробілу між словами.
+# Removing additional spaces from the text.
+new_adventures_of_tom_sayer = ' '.join(new_adventures_of_tom_sayer.split())
+_log.info(new_adventures_of_tom_sayer)
+
+# task 04
+# Виведіть, скількі разів у тексті зустрічається літера "h"
+# Count 'h' letter in new_adventures_of_tom_sayer variable and display result.
+letter_h = new_adventures_of_tom_sayer.count('h')
+_log.info('Letter "h" is found %s times in the text.', letter_h)
+
+# task 05
+# Виведіть, скільки слів у тексті починається з Великої літери?
+# Replacing " with empty spaces to prevent including " in words.
+# Creating list of words from the string with split function.
+words = new_adventures_of_tom_sayer.replace('"', '').split()
+# Creating new list of words with capitalized letter.
+# Calculate quantity of words with capitalized letter with function len().
+uppercase_words = len([word for word in words if word.istitle()])
+_log.info('There are %s title letter words in the text.', uppercase_words)
```

Comment:
Better to use f-string

Reply:
I receive this message from Pylint if I use f-string.
Use lazy % formatting in logging functionsPylint[W1203:logging-fstring-interpolation](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/logging-fstring-interpolation.html)